### PR TITLE
Implements StartTLS in HttpConnectSettings and NetworkHttpClient. (Followup)

### PR DESCRIPTION
**Pull request by dom96**
Corrected version of https://github.com/capnproto/capnproto/pull/1635.

For reference, we want to ensure:

* No API breakage
* That kj-http doesn't depend on kj-tls.

This new PR should now ensure this.

### Test Plan

```
bazel test @capnp-cpp//src/kj/compat:tls-test --config=asan --test_timeout 5
```

### CODE REVIEW

**Review by harrishancock — APPROVED** [review]

**Review by kentonv — COMMENTED** [review]
Sorry, I thought I posted this earlier but it was still pending.

File: c++/src/kj/compat/tls-test.c++ [hunk 1]
```diff
@@ -1149,6 +1151,44 @@ KJ_TEST("TLS receiver does not stall on hung client") {
   KJ_EXPECT(!extraAcceptPromise.poll(test.io.waitScope));
 }
 
+#if !_WIN32 // TODO: Investigate and fix issue on Windows.
```
Comment (kentonv) [inline c++/src/kj/compat/tls-test.c++:1154]: Can you elaborate? We are going to be supporting Windows soon so ignoring it here might be an issue.

**Comment by kentonv** [comment]
Looks good except we really shouldn't disregard Windows failures without at least understanding why they are happening. We are porting workerd to Windows so this could actually matter.

**Comment by dom96** [comment]
@kentonv I've got a task to follow up on the Windows aspect of this, not sure why it's failing and I'd have to switch workstations to investigate so I wanted to leave it for later.

**Comment by kentonv** [comment]
OK, as long as you plan to proactively follow up within the next few weeks.

**Review by kentonv — APPROVED** [review]

**Pull request merged**